src/sugar3/activity/activityhandle.py

```diff
@@ -15,47 +15,47 @@
 # Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 # Boston, MA 02111-1307, USA.
 
-"""
-STABLE.
-"""
+'''
+Provides a class for storing activity metadata such as activity id's,
+journal object id's,
+'''
 
 
 class ActivityHandle(object):
-    """Data structure storing simple activity metadata"""
+    '''
+    Data structure storing simple activity metadata
+    
+    Args:
+        activity_id (string): unique id for the activity to be
+        created
+        
+        object_id (string): identity of the journal object
+        associated with the activity.
+        
+        When you resume an activity from the journal
+        the object_id will be passed in. It is optional
+        since new activities does not have an
+        associated object.
+        
+        uri (string): URI associated with the activity. Used when
+        opening an external file or resource in the
+        activity, rather than a journal object
+        (downloads stored on the file system for
+        example or web pages)
+        
+        invited (bool): True if the activity is being launched for handling an invite
+        from the network
+    '''
 
     def __init__(self, activity_id=None, object_id=None, uri=None,
                  invited=False):
-        """Initialise the handle from activity_id
-
-        activity_id -- unique id for the activity to be
-            created
-        object_id -- identity of the journal object
-            associated with the activity. It was used by
-            the journal prototype implementation, might
-            change when we do the real one.
-
-            When you resume an activity from the journal
-            the object_id will be passed in. It's optional
-            since new activities does not have an
-            associated object (yet).
-
-            XXX Not clear how this relates to the activity
-            id yet, i.e. not sure we really need both. TBF
-        uri -- URI associated with the activity. Used when
-            opening an external file or resource in the
-            activity, rather than a journal object
-            (downloads stored on the file system for
-            example or web pages)
-        invited -- the activity is being launched for handling an invite
-                   from the network
-        """
         self.activity_id = activity_id
         self.object_id = object_id
         self.uri = uri
         self.invited = invited
 
     def get_dict(self):
-        """Retrieve our settings as a dictionary"""
+       '''Returns activity settings as a dictionary in format {activity_id:XXXX, object_id:XXXX, uri:XXXX, invited:BOOL}'''
```

Comment:
You've done something funny with the indent, as sugar crashes when this patch is applied:

```
Traceback (most recent call last):
  File "/usr/lib64/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/jarabe/main.py", line 66, in <module>
    from jarabe.view import keyhandler
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/jarabe/view/keyhandler.py", line 33, in <module>
    from jarabe.journal import journalactivity
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/jarabe/journal/journalactivity.py", line 32, in <module>
    from sugar3.activity import activityfactory
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/sugar3/activity/activityfactory.py", line 30, in <module>
    from sugar3.activity.activityhandle import ActivityHandle
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/sugar3/activity/activityhandle.py", line 59
    result = {'activity_id': self.activity_id,
    ^
IndentationError: unexpected indent
```

Also, please keep lines shorter than 80 charecters.  This one looks too long.